src/parameterizations/vertical/MOM_opacity.F90

```diff
@@ -254,6 +268,16 @@ subroutine opacity_from_chl(optics, sw_total, sw_vis_dir, sw_vis_dif, sw_nir_dir
 ! use the "blue" band in the parameterizations to determine the e-folding
 ! depth of the incoming shortwave attenuation. The red portion is lumped
 ! into the net heating at the surface.
+! Adding Ohlmann scheme. Needs sw_total and chl as inputs. Produces 2 penetrating bands.
+! This implementation follows that in CESM-POP using a lookup table in log10(chl) space.
+! The table is initialized in subroutine init_ohlmann and the coefficients are recovered
+! with routines lookup_ohlmann_swpen and lookup_ohlmann_opacity.
+! Note that this form treats the IR solar input implicitly: the sum of partioning
+! coefficients < 1.0. The remainder is non-penetrating and is deposited in first layer
+! irrespective of thickness. The Ohlmann (2003) paper states that the scheme is not valid
+! for vertcal grids with first layer thickness < 2.0 meters.
```

Comment:
2 comments:
1. Request to modify.
2. Question.
Thanks to @Hallberg-NOAA for alerting me about this addition from NCAR.


[1] Since this scheme was developed for top layer thickness < 2 m, there should be a call to `MOM_error` when that is not the case in which the model was configured at run time. This simple "internal self-consistency" is necessary to prevent misinterpretation by those who do not know (or care!) for _all_ details!
[2]  [Pp. 1344, eqn (8)](https://journals.ametsoc.org/view/journals/clim/16/9/1520-0442_2003_16_1337_orhicm_2.0.co_2.xml) includes a way to include solar zenith and cloudiness. They don't seem to be included **explicitly in this PR**, True? All that is being provided here are the coefficients in `init_ohlmann_table`?

Reply:
Thanks for these comments. 
1.) We can not add an error check in a straightforward manner because the first layer thickness is not readily available in MOM_opacity. The layer thickness is also dynamic, so would need to be checked at every grid point and every time step. We have added a note to use Ohlmann only if dz(1) > 2.0m  the get_param for "OPACITY_SCHEME".
2. The correction for zenith angle and cloud fraction is not included in this implementation. It is a second order effect as described in the paper. It is only non-zero where cloud fraction < 0.1 which is very rare over the ocean. The only region where these conditions frequently exist is the eastern Mediterranean.

Reply:
Thanks @fobryan3 for addressing both points - clarifying the scope of this implementation.

> We have added a note to use Ohlmann only if dz(1) > 2.0m the get_param for "OPACITY_SCHEME".
👍

Reply:
I believe that this comment has been addressed, so we can mark it as resolved.